GAT 2023 updates for Galaxy 23.0 (#3988)
galaxyproject/training-material: topics/admin/tutorials/customization-manual/tutorial.md

---
layout: tutorial_hands_on

title: "Customizing the look of Galaxy (Manual)"
questions:
  - How do I change Galaxy's front page?
  - How do I set a custom brand text?
  - How can I configure theming?
objectives:
  - Set a brand text for your Galaxy instance
  - Add a custom welcome page to Galaxy
  - Activate and customize themes
time_estimation: "45m"
key_points:
  - Set your Galaxy instance apart from others
  - Communicate what your Galaxy instance is about
  - Offer users more customization options using themes
contributors:
  - ElectronicBlueberry
requirements: []
subtopic: maintenance
---

# Customizing the look of Galaxy 

> Customizing your Galaxy instance makes it more recognizable at a glance, and can help communicate it's purpose to it's users.

Reviewer: ```suggestion
Customizing your Galaxy instance makes it more recognizable at a glance, and can help communicate its purpose to its users.
```

>
> 0. (optional) Set the location of your welcome page in `galaxy.yml`
> 
>    Under `galaxy:` add the following option:
>
>    ```yaml
>    welcome_url: /static/your-welcome-page.html
>    ```
>
>    Not setting this will cause the url to default to `/static/welcome.html`
>
> 1. Open `/static/welcome.html`, or create a new html file at the location set in Step 0
>
>    It is advisable to keep a copy of the default `welcome.html`, as it can serve as a useful reference.
>
> 2. Add some Content
>
>    Let's add a title and some text to our welcome page:
>
>    ```html
>    <!DOCTYPE html>
>    <html lang="en">
>        <head>
>            <meta charset="utf-8">
>        </head>
>        <body>
>            <h1>Welcome to the Galaxy Mars instance!</h1>
>            <p>The only Galaxy instance on mars.</p>
>        </body>
>    </html>
>    ```
>
>    This works the same as any other html page. You can add styles in the head, or import some scripts.
>    
>    Make sure your welcome page has exactly one `h1` element, which describes the page.
>    This will act as the heading for your start-page, which can help assistive technologies.
{: .hands_on}

> todo: add note about style/font

Reviewer: Bonus todo: could mention that it's convenient to make this an iframe like we do on .*

Author: Resolved

(figure out how to first...)

## Custom Masthead Theme

Galaxy's theming system allows you to easily change the color of your Masthead, to give your instance an even more distinct look.
You can even offer several options, to allow users to switch to the default if they prefer it, or another look all together.

> <hands-on-title>Configuring Themes</hands-on-title>
>
> 0. (optional) Set the location of the themes configuration `galaxy.yml`
> 
>    Under `galaxy:` add the following option:
>
>    ```yaml
>    themes_config_file: your-themes-file.yml
>    ```
>
>    This location is relative to the config directory.
>    If unset, the default file `themes_conf.yml` will be used, if present.
>

> > 1. Create you themes config file.

Reviewer: ```suggestion
> 1. Create your themes config file.
```

>

> >    Open `themes_conf.yml.sample`, and copy it's contents into a new file, either called `themes_conf.yml`, or the custom path you set in step 0.
> >    Do not copy the conf file and remove the ending, as this file is a symbolic link.
> >    Copying it copies a reference to the source file, not it's contents.

Reviewer: ```suggestion
>    Open `themes_conf.yml.sample`, and copy its contents into a new file, either called `themes_conf.yml`, or the custom path you set in step 0.
>    Do not copy the conf file and remove the ending, as this file is a symbolic link.
>    Copying it copies a reference to the source file, not its contents.
```

>
> 2. Understanding Themes
>

> >    The themes file can contain sever themes.
> >    Each theme is identified by it's id, eg `blue`, and followed by a set of rules, which style the client.
> >    The first theme in your theme file will be used as the default theme, which users see when visiting your instance,
> >    and which users which haven't logged in will see.

Reviewer: ```suggestion
>    The themes file can contain several themes.
>    Each theme is identified by its id, eg `blue`, and followed by a set of rules, which style the client.
>    The first theme in your theme file will be used as the default theme, which users see when visiting your instance,
>    and which users who haven't logged in will see.
```
